integration-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -293,7 +294,14 @@ private void scheduleNextChunk() {
         Chunk nextChunk = chunks.peek();
         if (nextChunk != null) {
           scheduled = true;
-          executor.schedule(dispatchTask, nextChunk.delayMicroseconds, TimeUnit.MICROSECONDS);
+          if (executor instanceof ScheduledExecutorService) {
```

Comment:
It allows testing a streaming response producer that does its work on the network thread using direct executor.  See the effect in NettyTransportIoEfficiencyTest

smallServerStreaming vs smallServerStreamingDirectExecutor